tests/general_checks.py

```diff
@@ -12,6 +13,7 @@ def general_output_checks(
     expected_results=None,
     verify_crs: bool = False,
     verify_attrs: bool = False,
+    expected_dims: list = [],
```

Comment:
```suggestion
    expected_dims: list = None,
```
See [here](https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument) for why this wants to be avoided!